src/main/java/algorithms/orthogonalRangeSearching/twoDim/OrthogonalRangeSearching.java

```diff
@@ -0,0 +1,314 @@
+     * @param yHigh  The upper bound of the X-coordinate range.
+     * @return The X-split node or null if not found.
+     */
+    public static RangeTreeNode<Integer[]> findYSplit(RangeTreeNode<Integer[]> root, int yLow, int yHigh) {
+        RangeTreeNode<Integer[]> v = root;
+
+        while (true) {
+            if (v == null) {
+                return null;
+            } else {
+                if (yHigh <= v.getVal()[1]) {
+                    if (v.getLeft() == null && v.getRight() == null) { // extra check since ysplit might be leaf node
+                        break;
+                    } else {
+                        v = v.getLeft();
+                    }
+                } else if (yLow > v.getVal()[1]) {
+                    v = v.getRight();
+                } else {
+                    break;
+                }
+            }
+        }
+        return v;
+    }
+
+    /**
+     * Performs a recursive traversal of the Range Tree and adds leaf node values to the result list.
+     *
+     * @param v      The current node being processed during traversal.
+     * @param result The list to store the values of leaf nodes encountered during traversal.
+     */
+    public static void allLeafTraversal(RangeTreeNode<Integer[]> v, List<Integer[]> result) {
+        if (v != null) {
+            if (v.getLeft() != null) {
+                allLeafTraversal(v.getLeft(), result);
+            }
+            if (v.getLeft() == null && v.getRight() == null) { // leaf
+                result.add(v.getVal());
+            }
+            if (v.getRight() != null) {
+                allLeafTraversal(v.getRight(), result);
+            }
+        }
+    }
+
+    /**
+     * Performs a left traversal of the Y-tree to find points within the specified range.
+     *
+     * @param v      The current node in the Y-tree.
+     * @param low    The lower bound of the Y-coordinate range.
+     * @param result A list to store the results.
+     */
+    public static void leftTraversal(RangeTreeNode<Integer[]> v, int low, List<Integer[]> result) {
+        if (v != null) {
+            if (v.getLeft() == null && v.getRight() == null) { // leaf
+                result.add(v.getVal());
+            } else {
+                if (low <= v.getVal()[1]) {
+                    leftTraversal(v.getLeft(), low, result);
+                    allLeafTraversal(v.getRight(), result);
+                } else { // definitely a qualifying leaf has to exist
+                    leftTraversal(v.getRight(), low, result);
+                }
+            }
+        }
+    }
+
+    /**
+     * Performs a right traversal of the Y-tree to find points within the specified range.
+     *
+     * @param v      The current node in the Y-tree.
+     * @param high    The upper bound of the Y-coordinate range.
+     * @param result A list to store the results.
+     */
+    public static void rightTraversal(RangeTreeNode<Integer[]> v, int high, List<Integer[]> result) {
```

Comment:
maybe can standardise the naming? It was named XRightTraversal. I might be missing smth, so feel free to share